### PR TITLE
Add GPT-4o Mini model to OpenAI provider

### PR DESCRIPTION
**Pull request by vishesh10**
### Background

This is in continuation to the https://github.com/Significant-Gravitas/AutoGPT/pull/7694 PR. 

Along with the addition of GPT-4o Mini support, the chromedriver issue is fixed which was being encountered in the https://github.com/Significant-Gravitas/AutoGPT/pull/7694.

### Changes 🏗️

* Updated the webdriver-manager version.
* Added the GPT-4o Mini model to the OpenAI provider in the openai.py file.
* Defined new model names GPT4_O_MINI_v1, GPT4_O_MINI_ROLLING, and GPT4_O_MINI.
* Set prompt token cost to 0.15/1,000,000 and completion token cost to 0.6/1,000,000 for GPT-4o Mini.
* Established a maximum token limit of 128,000 for GPT-4o Mini.
* Enabled support for the function call API for the GPT-4o Mini model.

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [x] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [x] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
enhancement, dependencies

___

### **Description**
- Added support for the GPT-4o Mini model in the OpenAI provider, including new model names and specifications.
- Defined prompt and completion token costs, and set a maximum token limit for GPT-4o Mini.
- Enabled the function call API for the GPT-4o Mini model.
- Updated the `webdriver-manager` dependency version to 4.0.2 to fix related issues.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>openai.py</strong><dd><code>Add GPT-4o Mini model support to OpenAI provider</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

forge/forge/llm/providers/openai.py

<li>Added new model names for GPT-4o Mini.<br> <li> Defined cost and token limits for GPT-4o Mini.<br> <li> Enabled function call API for GPT-4o Mini.<br> <li> Updated chat model mapping with GPT-4o Mini.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/8007/files#diff-8ad06a400003efdd02abe8e5983a6f44468fe9abb3cc89103030208033bcd9f5">+12/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Update webdriver-manager dependency version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

forge/pyproject.toml

- Updated `webdriver-manager` version to 4.0.2.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/8007/files#diff-43efe5f8aafe9a915d83f71d46e84d603a2be2f36566563db621a1e0c90dd14c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: forge/pyproject.toml [hunk 1]
```diff
@@ -54,7 +54,7 @@ tiktoken = ">=0.7.0,<1.0.0"
 toml = "^0.10.2"
 uvicorn = { extras = ["standard"], version = ">=0.23.2,<1" }
 watchdog = "4.0.0"
-webdriver-manager = "^4.0.1"
+webdriver-manager = "^4.0.2"
```
Comment (kcze) [inline forge/pyproject.toml:57]: Was this needed?

**Review by kcze — DISMISSED** [review]
LGTM, I checked that all numbers are correct (version, pricing, token count)

File: forge/pyproject.toml (hunk 1 above)
Comment (vishesh10) [reply forge/pyproject.toml:57]: Thanks for the review @kcze.
This was done to address the [chromedriver](https://github.com/Significant-Gravitas/AutoGPT/pull/7694#issuecomment-2271928384) issue.

**Comment by ntindle** [comment]
Sorry we kinda dropped the ball here, I'm going to close for now as the PR will need recreating. Ping me and I'll make sure it goes in if you want to recreate

**Pull request closed without merging**